### PR TITLE
UDN: Use managedTap instead passt

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
Use `managedTap` binding instead `Passt` for primary UDN.
It is registered as part of OVN kind.sh that we use in this repo to spin the cluster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Currently only IPv4 is supported.
A follow-up will add IPv6 support, by adding it to cloud-init, and deploying a dual stack cluster.
Once OVN use `l2bridge` naming we will adapt as well.

**Release note**:

```release-note
None
```

### CODE REVIEW

File: hack/cluster.sh
Comment (oshoval) [inline hack/cluster.sh:55]: TODO - drop

**Comment by oshoval** [comment]
Hi @qinqon @maiqueb 
once we want, we can switch to managedTap here (and then on CNAO when it tests it upstream as well)
just tests are changed on CNAO / here
(CNAO will also register only on upstream the binding / FGs via its upstream deployment scrips)

**Review by RamLavi — APPROVED** [review]
/lgtm

**Comment by oshoval** [comment]
@maiqueb @qinqon 
Can you please add it to your review queue ?
Thanks

**Review by maiqueb — COMMENTED** [review]
How can this work ? 

Where do we register the bindings ? 

Why are you dropping cloud init ?

**Comment by oshoval** [comment]
we register the binding on OVN k8s repo, that we use in this repo
we drop the cloud-init because as you also raised in the past (on slack, when you reviewed this gist [1]), this is not needed, it already happens automatically

[1] https://gist.github.com/RamLavi/91a25858a56f87e47039ceef99df662b

**Comment by maiqueb** [comment]
> we register the binding on OVN k8s repo, that we use in this repo we drop the cloud-init because as you also raised in the past (on slack, when you reviewed this gist [1]), this is not needed, it already happens automatically

Please add this info to the commit message.

> 
> [1] https://gist.github.com/RamLavi/91a25858a56f87e47039ceef99df662b

File: test/e2e/persistentips_test.go
Comment (qinqon) [inline test/e2e/persistentips_test.go:389]: for managedTap we need to activate dhcpv4 and dhcpv6.

File: test/e2e/persistentips_test.go
Comment (qinqon) [inline test/e2e/persistentips_test.go:405]: we still need cloudInit for managedTap.

File: test/e2e/persistentips_test.go
Comment (oshoval) [reply test/e2e/persistentips_test.go:389]: aint kubevirt assign ipv4 by default ? 
VM does get ipv4, and this what i remember for the past, wdyt 

ipv6 we don't test on git actions, (also on OVN we dont, because git actions doesnt support it), do we still want to add it ?

Comment (maiqueb) [reply test/e2e/persistentips_test.go:389]: wait. This depends on the cluster's configuration.
We need to introspect the cluster, figure out which IP families are enabled, and configure this accordingly. 

OR we can hard-code it to support whatever it is we're deploying by default in the e2e tests, but make a real good note about it, then open an issue to introspect in a follow-up PR.

Comment (qinqon) [reply test/e2e/persistentips_test.go:389]: > aint kubevirt assign ipv4 by default ? VM does get ipv4, and this what i remember for the past, wdyt ?
> 
> ipv6 we don't test on git actions, (also on OVN CI we dont have the cluster with ipv6, because git actions doesnt support it), do we still want to add it ? EDIT - hmm this one does have `.github/workflows/test.yml: KIND_IPV6_SUPPORT: "${{ matrix.ipfamily == 'IPv6' || matrix.ipfamily == 'dualstack' }}"`

That's not true, we run dual stack at ovn-k, and is ovn-k the one handling IPs.

Comment (oshoval) [reply test/e2e/persistentips_test.go:389]: best imo the 2nd option, have it on a follow-up

Comment (qinqon) [reply test/e2e/persistentips_test.go:389]: > wait. This depends on the cluster's configuration. We need to introspect the cluster, figure out which IP families are enabled, and configure this accordingly.
> 
> OR we can hard-code it to support whatever it is we're deploying by default in the e2e tests, but make a real good note about it, then open an issue to introspect in a follow-up PR.

We just acivate dhcp for both and that's all, if cluster is just ipv4 we will not have an ipv6 address.

Comment (oshoval) [reply test/e2e/persistentips_test.go:389]: i dont think we should add code that might not have an effect, better to have a task to investigate as Miguel said, what cluster we run here and only then add whatever config that is needed

Comment (oshoval) [reply test/e2e/persistentips_test.go:389]: btw does ipv6 is tested on OVN repo ? also something that worth checking
Once we have agreement on the direction i will do that, personally i prefer the one that doesnt have unused code and to open issue that is focused and changed whatever needed on all repos.
We can first finish the IPv4 flow on all the repos and then add the IPv6 tests according needs (and according what cluster we have).
Atm we are running IPv4 cluster on this repo.

**Comment by oshoval** [comment]
once we merge the new binding name on HCO, need to adapt it on OVN and here
(not because HCO affect this repos, but to be aligned of course)

File: test/e2e/persistentips_test.go
Comment (maiqueb) [reply test/e2e/persistentips_test.go:389]: > btw does ipv6 is tested on OVN repo ?

It surely is in some lanes. It's a configuration knob.

> Atm we are running IPv4 cluster on this repo.

So we'll need to change that. That means we're testing half the feature.

Sure, for now, you can move forward with ipv4 (since it appears it's the only thing we support).

But ensure you open a ticket to add IPv6 support to this repo.

Comment (oshoval) [reply test/e2e/persistentips_test.go:389]: > It surely is in some lanes. It's a configuration knob.

ye i meant if it is tested for managedTap itself on OVN

np will open Jira ticket
thanks

**Comment by oshoval** [comment]
Updated the PR description
I can return the cloud-init if desired, but it is not mandatory, the virt-launcher DHCP server already does it for IPv4,
maybe just once we have IPv6 we will need both ?

**Comment by maiqueb** [comment]
@qinqon any objections about merging this? It's blocking required bits at HCO. 
/lgtm

**Comment by oshoval** [comment]
hmm i had to do some leftovers, i need to recall them 
might be nits

**Comment by maiqueb** [comment]
> hmm i had to do some leftovers, i need to recall them might be nits
> 
> for example [#73 (comment)](https://github.com/kubevirt/ipam-extensions/pull/73#discussion_r1865922572) and to use `l2bridge` and update PR desc or so better to wait a bit please if possible

We don't have much time. We need this ASAP.

**Comment by oshoval** [comment]
rebased, disregard

**Comment by oshoval** [comment]
lets see it pass, i will check if more stuff is needed meanwhile

File: test/e2e/persistentips_test.go
Comment (oshoval) [reply test/e2e/persistentips_test.go]: done

**Comment by oshoval** [comment]
from quick check, all is done now
ptal

**Comment by oshoval** [comment]
@RamLavi 
would you able please to add the missing binding for CNAO as part of your ticket?
(update kubevirt CR with kubevirt FG and the binding registration as OVN / HCO does)
required since CNAO doesnt use `-ikv` that install kubevirt and does that by kind.sh of OVN

**Comment by oshoval** [comment]
/hold

it will fail
we need https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4854

or to revert here back to managedTap

**Comment by qinqon** [comment]
> /hold
> 
> it will fail we need [ovn-kubernetes/ovn-kubernetes#4854](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4854)
> 
> or to revert here back to the name `managedTap` whatever you prefer
> 
> note that HCO uses `l2bridge` (it is just the custom name of course, behind the scene both are internally `managedTap`) so best to do it right and one time

we patch replace managedTap after clonning ovn-kubernetes, is dirty but that will unblock this.

**Comment by oshoval** [comment]
> > /hold
> > it will fail we need [ovn-kubernetes/ovn-kubernetes#4854](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4854)
> > or to revert here back to the name `managedTap` whatever you prefer
> > note that HCO uses `l2bridge` (it is just the custom name of course, behind the scene both are internally `managedTap`) so best to do it right and one time
> 
> we patch replace managedTap after clonning ovn-kubernetes, is dirty but that will unblock this.

i dont like it, it will require fix later anyhow,
why not to just either wait for 4854 to be merged, or to use here `managedTap` which is already registered
and once 4854 is merged to adapt, easy to adapt

**Comment by oshoval** [comment]
just rebased

**Comment by oshoval** [comment]
just rebased

**Comment by oshoval** [comment]
updated name to managedTap until OVN adapt it to l2bridge
as discussed offline

/hold cancel

**Comment by maiqueb** [comment]
Let's get this in today, and adapt once ovn-k deploys l2bridge via https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4854

/lgtm

@qinqon any objection ?

**Comment by oshoval** [comment]
@RamLavi 
i will issue a release ok ?
and then CNAO can be adapted to use managedTap

maybe just bump + https://github.com/kubevirt/cluster-network-addons-operator/pull/1919 is needed

**Comment by oshoval** [comment]
since a tag was already created yesterday without this PR, and this one is needed i created a new one
v0.1.9-alpha

**Pull request merged**
